clang/lib/DPCT/Utility.cpp

```diff
@@ -4736,5 +4736,55 @@ std::string appendPath(const std::string &P1, const std::string &P2) {
   llvm::sys::path::append(TempPath, P2);
   return TempPath.str().str();
 }
+
+std::error_code createDirectories(const clang::tooling::UnifiedPath &FilePath,
+                                  bool IgnoreExisting) {
+  std::error_code EC = llvm::sys::fs::create_directories(
+      FilePath.getCanonicalPath(), IgnoreExisting);
+  if ((bool)EC) {
+    std::string ErrMsg =
+        "[ERROR] Create Directory : " + FilePath.getPath().str() +
+        " fail: " + EC.message() + "\n";
+    clang::dpct::PrintMsg(ErrMsg);
+  }
+  return EC;
+}
+
+// std::ios::binary prevents ofstream::operator<< from converting \n to \r\n
+// on windows.
+bool writeDataToFile(const std::string &FileName, const std::string &Data) {
+  auto FileStream =
+      std::make_unique<CheckedOfstream>(FileName, std::ios::binary);
+  *FileStream << Data;
+  FileStream->close();
+  if (FileStream->fail()) {
```

Comment:
will it duplicate with ~CheckedOfstream()?

Reply:
The ~CheckedOfstream() uses to work with the llvm::raw_os_stream. And writeDataToFile uses to write the data to std::ofstream